### DISCUSSION TITLE
Attributes: what do the source and selector properties do?

### DISCUSSION THREAD

**Discussion by mburridge** [WordPress/developer-blog-content#138] [Topic Ideas]
Block attributes are typically stored in the comment delimiter, and the values are fetched from there when the block is parsed to be displayed in the editor. But did you know that attribute values can also be stored in the block content? You can use the `source` and `selector` properties to determine where in the saved block the attribute values are fetched from at parse time.

This post will be an introduction to using and saving attributes - basically a friendlier and more tutorial-based version of the content in the [Attributes page](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/) in the handbook, and with more examples.

**Comment by justintadlock** [comment]
Sounds like a good topic, @mburridge. We could definitely use a post exploring this.

**Comment by bph** [comment]
The topic has been approved. The discussion is now locked,
Please continue the conversation on the issue
- #147

**Comment by mburridge** [comment]
Now published: https://developer.wordpress.org/news/2023/09/understanding-block-attributes/
